### PR TITLE
STYLE: fix pylint redefined-outer-name warning for file

### PR DESCRIPTION
**Pull request by angularOcean**
Associated with [#49656](https://github.com/pandas-dev/pandas/issues/49656).
Fixes pylint warnings for `redefined-outer-name` on the following files:

`pandas/core/tools/datetimes.py`

- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).

### CODE REVIEW

File: pandas/core/tools/datetimes.py [hunk 1]
```diff
@@ -293,15 +293,17 @@ def _convert_and_box_cache(
...
+) -> Index:
     """
     Return results from array_strptime if a %z or %Z directive was passed.
 
     Parameters
     ----------
     result : ndarray[int64]
         int64 date representations of the dates
-    timezones : ndarray
+    timezone_arr : ndarray
```
Comment (MarcoGorelli) [inline pandas/core/tools/datetimes.py:306]: you shouldn't rename function arguments

rather, you could add a `as libtimezones` to the `timezones` import

**Review by MarcoGorelli — CHANGES_REQUESTED** [review]

File: pandas/core/tools/datetimes.py (hunk 1 above)
Comment (angularOcean) [reply pandas/core/tools/datetimes.py]: Ah my mistake!  Reverted function name change and renamed timezones import to libtimezones as suggested.

**Comment by MarcoGorelli** [comment]
no need to keep updating the branch - this is probably fine, let's let the checks run and I'll merge later

**Review by MarcoGorelli — APPROVED** [review]
thanks @angularOcean

**Comment by angularOcean** [comment]
You are welcome! Thank you for the opportunity to contribute!

**Pull request merged**